### PR TITLE
[Pibbble] [#106189306] Project likes feature

### PR DESCRIPTION
**Pull request by wilmkr**
Here's the project like feature. We need to run the create_projects_likes_table migration for this feature to work.

### CODE REVIEW

File: app/Http/Controllers/ProjectLikesController.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+     * @param  $projectID       [the id of the project liked]
+     * @return response         [JSON response returned to the AJAX calling function]
+     */
+    public function like($projectID)
+    {
+        $userID = Auth::user()->id;
+
+        $result = ProjectLikes::where('project_id', '=', $projectID)->where('user_id', '=', $userID)->first();
+
+        if(is_null($result)) {
```
Comment (unicodeveloper) [inline app/Http/Controllers/ProjectLikesController.php]: Fix this

File: app/Http/Controllers/ProjectLikesController.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            $projectlikes = new ProjectLikes();
+            $projectlikes->project_id = $projectID;
+            $projectlikes->user_id = $userID;
+            $projectlikes->save();
+        }
```
Comment (unicodeveloper) [inline app/Http/Controllers/ProjectLikesController.php]: Fix

File: app/Http/Controllers/ProjectLikesController.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        else {
+            $theLike = ProjectLikes::find($result->id);
+            $theLike->delete();
+        }
+
+        $likes = DB::table('projects_likes')->where('project_id', '=', $projectID)->count();
+
+        return response()->json($likes);
+    }
+}
```
Comment (unicodeveloper) [inline app/Http/Controllers/ProjectLikesController.php]: **Code Quality - Nitpick**

`else` should be on the same line as the closing brace of `if`

line 31 and 32 can be squashed into one line like so:

``` php
  ProjectLikes::find($result->id)->delete();
```

File: app/ProjectLikes.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    public function project()
+    {
+        return $this->belongsTo('Pibbble\Project');
+    }
+
+    public function user()
+    {
+        return $this->belongsTo('Pibbble\User');
+    }
+}
```
Comment (unicodeveloper) [inline app/ProjectLikes.php]: Fix

File: database/migrations/2015_12_09_120329_create_project_likes_table.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('projects_likes', function (Blueprint $table) {
+        $table->increments('id');
```
Comment (unicodeveloper) [inline database/migrations/2015_12_09_120329_create_project_likes_table.php]: Line 16 is badly indented. Please fix

File: database/migrations/2015_12_09_120329_create_project_likes_table.php [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        $table->integer('project_id')->unsigned();
+        $table->integer('user_id')->unsigned();
```
Comment (unicodeveloper) [inline database/migrations/2015_12_09_120329_create_project_likes_table.php]: Fix indentation

File: database/migrations/2015_12_09_120329_create_project_likes_table.php [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        $table->timestamps();
```
Comment (unicodeveloper) [inline database/migrations/2015_12_09_120329_create_project_likes_table.php]: Badly indented..Fix :red_circle:

File: database/migrations/2015_12_09_120329_create_project_likes_table.php [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        $table->foreign('project_id')->references('id')->on('projects')->onDelete('cascade')->onUpdate('cascade');
```
Comment (unicodeveloper) [inline database/migrations/2015_12_09_120329_create_project_likes_table.php]: :red_circle: Fix

File: public/js/like.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
+function like(projectID, count, ajaxResponse, likeLink)
+{
+    //alert(likeLink);
+    if (window.XMLHttpRequest) {
+        xmlhttp = new XMLHttpRequest();
```
Comment (unicodeveloper) [inline public/js/like.js:5]: **Code Quality - Nitpick**

Please fix all badly indented migration files

**Comment by unicodeveloper** [comment]
@andela-womokoro why is this still pending?

**Comment by unicodeveloper** [comment]
@andela-womokoro have you fixed all issues pending here?

**Comment by wilmkr** [comment]
The issues are much even though they are minor. I'm working on them. Will be through soon.

**Comment by unicodeveloper** [comment]
Please, when will you be through with resolving the issues?

**Comment by wilmkr** [comment]
Latest wednesday. But i'm aiming to wrap it up today.

**Comment by wilmkr** [comment]
Prosper I have addressed the issues.

**Pull request merged**